userspace/libscap/engine/gvisor/scap_gvisor.cpp

```diff
@@ -377,4 +452,161 @@ int32_t engine::next(scap_evt **pevent, uint16_t *pcpuid)
     return SCAP_TIMEOUT;
 }
 
+std::vector<std::string> engine::runsc(char *argv[])
+{
+	std::vector<std::string> res;
+	int pipefds[2];
+
+	int ret = pipe(pipefds);
+	if(ret)
+	{
+		return res;
+	}
+
+	int pid = fork();
```

Comment:
Is this sort of manually re-creating `system()`?
Can't we use `popen` instead?

At the very least, can we use `vfork` here? Fork is super heavy and i do not think that we need to copy everything in the new process just to call exec.

Reply:
Hi Fede! Both `system` and `popen` use a shell to execute the command and I wanted to avoid that. Btw, I agree with you on using `vfork` so I have added it, thank you